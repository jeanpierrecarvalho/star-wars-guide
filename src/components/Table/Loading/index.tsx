import React from "react";

const Loading: React.FC = () => {
  return (
    <div
      className="z-10 absolute inset-0 flex justify-center items-center bg-black bg-opacity-60"
      role="status"
      aria-live="polite"
    >
      <span className="text-xl text-yellow-500">
        Patience you must have, my young Padawan...
      </span>
    </div>
  );
};

export default Loading;
