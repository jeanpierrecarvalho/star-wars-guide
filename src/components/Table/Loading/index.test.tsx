import { render } from '@testing-library/react'
import { describe, it, expect } from 'vitest'

import Loading from '@/components/Table/Loading'

describe('Loading Component', () => {
  it('matches the snapshot', () => {
    const { container } = render(<Loading />)
    expect(container).toMatchSnapshot()
  })
})
