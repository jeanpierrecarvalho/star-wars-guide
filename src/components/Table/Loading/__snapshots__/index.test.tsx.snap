// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading Component > matches the snapshot 1`] = `
<div>
  <div
    class="z-10 absolute inset-0 flex justify-center items-center bg-black bg-opacity-60"
  >
    <span
      class="text-xl text-yellow-500"
    >
      Patience you must have, my young Padawan...
    </span>
  </div>
</div>
`;
